[12.0] Refactor code and introduce pattern.config (inherit of ir.export)

kevinkhao opened the pull request.

kevinkhao committed:
[REF] Change notification system

kevinkhao commented:
Contains https://github.com/shopinvader/pattern-import-export/pull/52

kevinkhao committed:
[REF] Create enqueue function to avoid duplicated code

kevinkhao committed:
[REF] Refactor everything relevant to pattern_import_export on ir.exports into a new pattern.config model that delegation-inherits ir.exports

kevinkhao committed:
[IMP] Improve and fix views for pattern config

kevinkhao committed:
[FIX] Fix broken onchanges due to pattern.config delegated-inheritance of ir.exports

kevinkhao committed:
[FIX] Add migration scripts

sebastienbeau commented on pattern_import_export/models/pattern_file.py:
```diff
@@ -17,4 +19,94 @@ class PatternFile(models.Model):
+    )
+
+    @api.model
+    def create(self, vals):
```
Wrong inherit vals can be a list, please use new API

sebastienbeau commented on pattern_import_export/models/pattern_file.py:
```diff
@@ -17,4 +19,94 @@ class PatternFile(models.Model):
+        return result
+
+    def _notify_user(self):
+        import_or_export = "Import" if self.kind == "import" else "Export"
```
This key should be translatable

sebastienbeau commented on pattern_import_export/models/pattern_file.py:
```diff
@@ -17,4 +19,94 @@ class PatternFile(models.Model):
+        web = "/web#"
+        args = [
+            "action="
+            + str(self.env.ref("pattern_import_export.action_pattern_file_imports").id),
```
action can be a string with the xml id

sebastienbeau commented on pattern_import_export/models/pattern_file.py:
```diff
@@ -17,4 +19,94 @@ class PatternFile(models.Model):
+                ),
+                sticky=True,
+            )
+        else:
```
use elif self.state = "done"
We do no want to notify pending case ;)

kevinkhao committed:
[FIX] Misc. fixes
correct create() to use new api
add translatability to a string
change action to xmlid for notification message
no notification on pending pattern_file's
make travis happy

sebastienbeau commented on pattern_import_export/migrations/12.0.4.0.0/post-migrate.py:
```diff
@@ -0,0 +1,53 @@
+        env.cr.execute(
+            """
+            UPDATE ir_exports_line
+            SET sub_pattern_config_id = pattern_export_id
```
not sure that this is correct.
If you have an existing ir.export (with id = 1) that is not a pattern.
Then the id 1 of pattern.config will not match the id of the ir.export
id of ir.export != id of pattern.config

sebastienbeau commented on pattern_import_export_csv/migrations/12.0.2.0.0/post-migrate.py:
```diff
@@ -0,0 +1,27 @@
+        )
+        env.cr.execute(
+            """
+                UPDATE pattern_config SET {} FROM ir_exports
```
I prefers to avoid adding logic here

```
                 UPDATE pattern_config
                 SET 
                         export_format=ir_exports.export_format,
                         csv_value_delimiter = ir_exports.csv_value_delimiter,
                         csv_quote_character = ir_exports.csv_quote_character
                 FROM ir_exports
                 WHERE export_id = ir_exports.id
```

It's faster to read and less line of code ;)

sebastienbeau committed:
[REF] simplify migration script

sebastienbeau commented:
ocabot merge nobump

sebastienbeau commented:
/ocabot merge nobump
